### ISSUE TITLE
'PotentialBond' comparison raises TypeError

### ISSUE THREAD

**Issue by bisdan** [Electrostatics/pdb2pqr#52]
Hello PDB2PQR Team,

during protonation with the flag --protonate-all the following TypeError is raised:

`apbs-pdb2pqr/pdb2pqr/pdb2pqr/utilities.py, line 26, in sort_dict_by_value:
items.sort()
TypeError: '<' not supported between instances of 'PotentialBond' and 'PotentialBond'`

Since you are sorting tuples at this point in the code this might be interesting to you: [https://stackoverflow.com/questions/5292303/how-does-tuple-comparison-work-in-python](url)

I think the TypeError appears when the first element (distance in your case) of some pair of tuples is equal. Then the second element will be compared. Since the second element is an instance of 'PotentialBond' and the comparison operator is not implemented an error is raised.

Changing `items.sort()` to `items.sort(key = lambda x: x[0])` in `apbs-pdb2pqr/pdb2pqr/pdb2pqr/utilities.py, line 26` works. Does this alter the intended behavior?

Cheers,

Daniel

**Comment by davipatti** [comment]
I've also been running into this error. For example:

~~~bash
$ (pdb2pqr-env) pdb2pqr --ff amber test.pdb test.pqr

[...]

TypeError: '<' not supported between instances of 'PotentialBond' and 'PotentialBond'
~~~

Here is [test.pdb](https://gist.githubusercontent.com/davipatti/329a98f6b688bce9bb704cd29322bd5c/raw/48ba9039e87b5ff7fac135e816a24c31e44c6b78/test.pdb)

Cheers
David

**Comment by sobolevnrm** [comment]
Hi @davipatti and @bisdan -- can you please let me know which version of PDB2PQR you are using?  That will help us determine where to apply the suggested patches.  Thanks!

**Comment by davipatti** [comment]
Sorry for my slow response - I've been away for a bit.

I'm getting this error with the most recent commit (29a8675596f6d6ba65a71eceb0e028689f733fba).

**Comment by sobolevnrm** [comment]
This seems to have been fixed -- I can't reproduce the error with 012ab94.
